feature/google/src/unityIAP/java/com/revenuecat/purchases/google/storeProductConversions.kt

```diff
@@ -1,19 +1,52 @@
 package com.revenuecat.purchases.google
 
 import com.android.billingclient.api.SkuDetails
+import com.revenuecat.purchases.ProductType
+import com.revenuecat.purchases.models.Price
+import com.revenuecat.purchases.models.PricingPhase
+import com.revenuecat.purchases.models.PurchaseOption
+import com.revenuecat.purchases.models.RecurrenceMode
 import com.revenuecat.purchases.models.StoreProduct
 import org.json.JSONObject
 
 private fun SkuDetails.toStoreProduct() =
     BC4StoreProduct(
         sku,
         type.toRevenueCatProductType(),
-        if (type == ProductType.INAPP) Price(price, priceAmountMicros, priceCurrencyCode) else null,
+        createPrice(),
         title,
         description,
         subscriptionPeriod.takeIf { it.isNotBlank() },
-        freeTrialPeriod.takeIf { it.isNotBlank() },
+        createPurchaseOptions(),
+        freeTrialPeriod,
+        introductoryPrice,
+        introductoryPriceAmountMicros,
+        introductoryPricePeriod,
+        introductoryPriceCycles,
         iconUrl,
-        JSONObject(originalJson),
-        this
-    )
\ No newline at end of file
+        originalPrice,
+        originalPriceAmountMicros,
+        JSONObject(originalJson)
+    )
+
+private fun SkuDetails.createPrice(): Price? {
+    return if (type.toRevenueCatProductType() == ProductType.INAPP)
+        Price(price, priceAmountMicros, priceCurrencyCode)
+    else null
+}
+
+private fun SkuDetails.createPurchaseOptions(): List<PurchaseOption> {
```

Comment:
@swehner let me know your thoughts on this... 

i'm thinking for BC4 store products it's a little messy. we want to only have one PurchaseOption, because there's only one purchaseable entity and the developer doesn't have control over a free trial/intro price being applied. For that reason, I'm hesitant to put the free trial/intro price into the pricing phases -- we wouldn't know WHICH to include or what order...

so i went with making a single pricing phase for the base sub, and then leaving all of the free trial/intro price-specific fields like before.

we could maybe unify interfaces further by just having BC4StoreProduct have introPricePricingPhases: List<PricingPhases> or something, but I don't think they can live in the purchase option

Reply:
another thought -- we've discussed somehow surfacing free trial/intro price eligiblity before, right @aboedo ? if we did that, we could neatly only pull those the user is eligible for into the purchaseoption... 

but i assume that's out of scope for this release, and i'm not sure we can even 💯 guarantee we can deduce it correctly

Reply:
yeah, intro eligibility isn't needed for v1 I'd imagine. FWIW for iOS we have 2 ways of obtaining it: 
- checkIntroEligibility which takes in products and returns a dict of product -> eligibility
- in each product, there's an isEligibleForFreetrial property

Those aren't the exact names but conceptually that's it